classes/image_types_ostree.bbclass

```diff
@@ -5,14 +5,12 @@ do_image_ostree[depends] += "ostree-native:do_populate_sysroot \
                         coreutils-native:do_populate_sysroot \
                         unzip-native:do_populate_sysroot \
                         virtual/kernel:do_deploy \
-                        ${OSTREE_INITRAMFS_IMAGE}:do_image_complete"
+"
```

Comment:
Are you sure this isn't required?  I got a failure when building of `cp: cannot stat '…/tmp/deploy/images/qemux86-64/initramfs-ostree-image-qemux86-64.cpio.gz': No such file or directory
`
I hadn't seen that failure before, so it might be a race condition.

Reply:
Hmm, this shouldn't happen even if you are not using FIT.
initramfs build is always ordered before kernel deploy task (in kernel.bbclass) and on the previous line we depend on kernel deploy task.

Is the image name correct? Maybe it is altered in some way for qemu..

Reply:
Weird. With a clean tmp directory it reliably fails with because `initramfs-ostree-image-qemux86-64.cpio.gz` is missing. Adding `${INITRAMFS_IMAGE}:do_image_complete` fixes it.

However, the only difference between the output of `bitbake -g core-image-minimal:do_image_ostree` with and without the change is an extra direct edge from `core-image-minimal.do_image_ostree` to `initramfs-ostree-image.do_image_complete`. In theory this shouldn't matter because there is a indirect dependency between the two already. In practice it appears there is something more subtle going on.